types/wordpress__blocks/index.d.ts

```diff
@@ -503,3 +504,20 @@ export type Transform<T extends Record<string, any> = Record<string, any>> =
     | TransformPrefix<T>
     | TransformRaw<T>
     | TransformShortcode<T>;
+
+export type BlockVariationScope = 'block' | 'inserter' | 'transform';
+
+export interface BlockVariation<TAttributes extends Record<string, any> = any> {
```

Comment:
although we do have `TAttributes` in `wordpress__blocks/api/registration` we don't otherwise have a precedent for prefixing generic type parameters with a letter. I'm definitely in the camp of wanting to avoid them as I see them largely as a unnecessary relic of string-based typing styles.

How do you feel about naming it simply `Attributes` or `BlockAttributes`?

Reply:
Hm, what do you think if we go with just `T` to follow how generics are used in the whole file? I'll push it like that with the other change and see how it feels.

I tend to go with more explicit naming for generic types as well but I feel like better to just keep it one style in here.

Reply:
to me `T` or `Attributes` is better than `TAttributes`.

I tend to lean towards semantic names when there's a clear semantic role and generic names when that's absent. for example, `Array<T>` is great because `Array` doesn't and should care what `T` is, beyond certain `extends` restrictions (which `Array` doesn't particularly have). In this case there's a more compelling argument that we don't want to simply communicate, "we just want any old key/value store," but instead that we want to communicate, "we are looking for block attributes here."

It's not something I would want to stand in the way of this patch, but you can see if that makes sense to you. I would probably pick `Attributes` but consistency has its value, if using `T` is _consistent_ with the rest of the module in that all generic parameters hold generic names when they intend to communicate specific semantic meaning as `BlockVariation` in fact does.